packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -25,6 +25,18 @@ export const SelectionHandler = (
 
   const setFilter = (filter?: Filter) => currentFilter = filter;
 
+  let currentAnnotatingEnabled = true;
+
+  const setAnnotatingEnabled = (enabled: boolean) => {
+    currentAnnotatingEnabled = enabled;
+    onSelectionChange.clear();
+
+    if (!enabled) {
+      currentTarget = undefined;
+      lastPointerDown = undefined;
+    }
+  };
```

Comment:
We must clean the `onSelectionChange` debounced call to prevent it from running on obsolete values from the closure. Otherwise, the `selectionchange` event will still get processed in the deferred call, even though the `currentAnnotatingEnabled` became `false`.

Reply:
I wonder how crucial that is though. Would `annotatingEnabled` get called during a selection in practice? The normal use cases I would see is:
- Setting it once on page init (because the user is getting a page served in "read-only mode")
- "Locking a page" interactively, by the user clicking a "Lock" button. (In which case there wouldn't be a selection.)